packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```diff
@@ -40,16 +41,32 @@ const {ReactShouldWarnActingUpdates} = ReactSharedInternals;
 // this implementation should be exactly the same in
 // ReactTestUtilsAct.js, ReactTestRendererAct.js, createReactNoop.js
 
-// we track the 'depth' of the act() calls with this counter,
-// so we can tell if any async act() calls try to run in parallel.
-let actingUpdatesScopeDepth = 0;
+let hasWarnedAboutMissingMockScheduler = false;
+const flushWork =
+  Scheduler.unstable_flushWithoutYielding ||
+  function() {
+    if (hasWarnedAboutMissingMockScheduler === false) {
+      warningWithoutStack(
+        null,
+        'Starting from React v17, the "scheduler" module will need to be mocked ' +
+          'to guarantee consistent behaviour across tests and browsers. To fix this, add the following ' +
+          "to the top of your tests, or in your framework's global config file -\n\n" +
+          'As an example, for jest - \n' +
+          "jest.mock('scheduler', () => require.requireActual('scheduler/unstable_mock'));\n\n" +
+          'For more info, visit https://fb.me/react-mock-scheduler',
+      );
+      hasWarnedAboutMissingMockScheduler = true;
+    }
 
-function flushEffectsAndMicroTasks(onDone: (err: ?Error) => void) {
+    while (flushPassiveEffects()) {}
+  };
+
+function flushWorkAndMicroTasks(onDone: (err: ?Error) => void) {
```

Comment:
How would this actually flush micro tasks (in an env like Jest)?

Reply:
that's what enqueueTask is for https://github.com/facebook/react/blob/master/packages/shared/enqueueTask.js, it waits for a so-called 'macro' task before continuing. Any specific concern?

Reply:
My concern is that if nothing tells Jest to advance it's timers, that macro task will never fire, and the test will just time out.